src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -2429,7 +2459,93 @@ private void copyPC2Sandbox() throws Exception {
+            
+        //check the OS to be sure we support interactive problems
+        if (OSCompatibilityUtilities.isRunningOnWindows() && !debugAllowInteractiveInvocationOnWindows) {
+            
+            log.severe("Attempt to execute a problem configured with an interactive validator on a Windows system: not supported");
+            throw new Exception ("Interactive Validators are not supported on Windows OS");
+            
+        } else {
+            
+            //OS supported (non-Windows values of osName could be "Linux", "SunOS", "FreeBSD", and "Mac OS X", all of which should work)
+                
+            log.info("Copying PC2 interactive validator scripts into Execute directory");
+            try {
+                //copy the PC2 interactive validator scripts into the execution directory
+                copyPC2InteractiveValidatorScripts();
+                createValidatorProgram();
+            } catch (Exception e) {
+                log.severe("Unable to copy PC2 interactive validator scripts to execute directory; cannot execute submission");
+                throw e;
+            }
+        }
+        
+        //Problem has a properly-configured interactive validator and we're not running a Team client
+        return;
+    }
 
+    /**
+     * Copies the PC2 interactive validator script files into the execution directory.
+     * 
+     * @throws Exception if creation of the script files in the execution directory failed.  
+     *          The Exception which is thrown is whatever Exception occurred during execution
+     *          of the file copy operation.
+     */
+    private void copyPC2InteractiveValidatorScripts() throws Exception {
+        
+        ArrayList<String> scriptNames = new ArrayList<String>();
+       
+        //If a sandbox is being used, the interactive validator sandbox script was copied already by setupSandbox(). If not
+        // using a sandbox, we have to copy the non-sandbox version of the interactive validator script here.
+        SandboxType sbType = problem.getSandboxType();
+        if (sbType == SandboxType.NONE) {
+            scriptNames.add(Constants.PC2_INTERACTIVE_NAME);
+        }
+        // Add in the PC2 final validator that validates the results of the interactive validator
+        scriptNames.add(Constants.PC2_INTERACTIVE_VALIDATOR_NAME);
+        
+        //use the VersionInfo class to get the PC2 installation directory
+        VersionInfo versionInfo = new VersionInfo();
+        String home = versionInfo.locateHome();      
+        
+        // Now, copy the files one at-a-time from the list
+        for(String scriptName : scriptNames) {
+            //point to the file that we want to create
+            String targetFileName = prefixExecuteDirname(scriptName);
+    
+            //point to the PC2 Internal Sandbox file (under "/sandbox" in the home, i.e. installation, directory)
```

Comment:
I think this comment needs to be updated; I don't think there is an longer a "/sandbox" folder...

Reply:
Comment updated.  Changed to /scripts.